Dj setup

riskarsh opened the pull request.

riskarsh committed:
Added .gitignore

riskarsh committed:
Django setup.

Setup python virtual environment for python dependencies

Basic setup for django app

riskarsh committed:
Ignore db file

maxking commented on .gitignore:
```diff
@@ -0,0 +1,5 @@
+venv/
+*.pyc
+*.sqlite3
+
+
```
Extra newlines at the end of the file, please remove.

maxking commented on requirements.txt:
```diff
@@ -0,0 +1,3 @@
+Django
+iPython
+
```
Same here, extra empty line.

maxking reviewed (changes requested):
Minor changes requested, otherwise, looks good to me.

maxking commented:
Also, please change the title of the Pull Request(PR) and the description to be more descriptive about the changes you are making.

riskarsh committed:
Added the User and Tweet models.
Added views to sort all tweets and tweets by users.
Added urlpatterns for it.

riskarsh committed:
Removed extra lines

maxking commented on twitter/twitter/urls.py:
```diff
@@ -0,0 +1,22 @@
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('twitter/',include("twitterapp.urls")),
```
Use either single quote or double quotes everywhere for consistency. 
```suggestion
    path('twitter/',include('twitterapp.urls')),
```

maxking commented on twitter/twitterapp/models.py:
```diff
@@ -0,0 +1,12 @@
+from django.db.models.base import Model
+
+# Create your models here.
+class User(models.Model): 
```
Add a class level docstring to document what this model represents.

For example, something like:

```
class User(models.Model): 
    """User represents an account in our system.

    A user is an owner of the tweets in the system and is considered
    as the author. Each person who signs up has a corresponding 
    User in the database.
    """

```

maxking commented on twitter/twitterapp/models.py:
```diff
@@ -0,0 +1,12 @@
+    name = models.CharField(max_length=50)
+    joined_date = models.DateTimeField('Acc creation date')
+
+class Tweet(models.Model):
```
Add a docstring for this Model too.

maxking commented on twitter/twitterapp/models.py:
```diff
@@ -0,0 +1,12 @@
+# Create your models here.
+class User(models.Model): 
+    name = models.CharField(max_length=50)
+    joined_date = models.DateTimeField('Acc creation date')
```
Try to always use proper English words to avoid any confusion.

```suggestion
    joined_date = models.DateTimeField('Account creation date')
```

maxking commented on twitter/twitterapp/views.py:
```diff
@@ -0,0 +1,18 @@
+from django.http import HttpResponse
+
+# Create your views here.
+def all_tweets(request):
```
Add a docstring for the view for documentation.

It is also useful to write the URL that this function will serve for future documentation.
```suggestion
def all_tweets(request):
    """Get all the tweets in the system.
    
    Since this returns everything from the database, this URL should be
    private or removed in future as it will cause very high load on the system.
    
    /twitter
    """
```

maxking commented on twitter/twitterapp/views.py:
```diff
@@ -0,0 +1,18 @@
+    tweets = Tweet.objects.all()
+    output = ""
+    for tweet in tweets:
+        output += "[{}] @{} - {}\n".format(tweet.date,tweet.user.name,tweet.content)
```
```suggestion
        output += "[{}] @{} - {}\n".format(tweet.date, tweet.user.name, tweet.content)
```

Add Empty space after comma.

maxking commented on twitter/twitterapp/views.py:
```diff
@@ -0,0 +1,18 @@
+    output = ""
+    for tweet in tweets:
+        output += "@{} - {} ".format(tweet.user.name,tweet.content)
+    return HttpResponse(output)    
```
```suggestion
    return HttpResponse(output)

```
Remove extra whitespace at the end.

maxking commented on twitter/twitterapp/views.py:
```diff
@@ -0,0 +1,18 @@
+    tweets = Tweet.objects.filter(user__id=user_id)
+    output = ""
+    for tweet in tweets:
+        output += "@{} - {} ".format(tweet.user.name,tweet.content)
```
```suggestion
        output += "@{} - {} ".format(tweet.user.name, tweet.content)
```
Empty space after comma.

maxking commented on twitter/twitterapp/views.py:
```diff
@@ -0,0 +1,18 @@
+        output += "[{}] @{} - {}\n".format(tweet.date,tweet.user.name,tweet.content)
+    return HttpResponse(output)
+
+def user_tweets(request,user_id):
```
Add docstring similar to `all_tweets`.

riskarsh committed:
Update twitter/twitter/urls.py

Co-authored-by: Abhilash Raj <maxking@users.noreply.github.com>

riskarsh committed:
Created a form.

- Form for creating a new tweet.
- Created a view for new tweet form the all tweets page.
- Created a page for viewing user tweets.
